web/src/lib/utils/byte-units.ts

```diff
@@ -8,23 +18,10 @@
  * @param maxPrecision maximum number of decimal places, default is `1`
  * @returns size (number) and unit (string)
  */
-export function getBytesWithUnit(bytes: number, maxPrecision = 1): [number, string] {
-  const units = ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB'];
-
-  let magnitude = 0;
-  let remainder = bytes;
-  while (remainder >= 1024) {
-    if (magnitude + 1 < units.length) {
-      magnitude++;
-      remainder /= 1024;
-    } else {
-      break;
-    }
-  }
-
-  remainder = Number.parseFloat(remainder.toFixed(maxPrecision));
+export function getBytesWithUnit(bytes: number, maxPrecision = 1): [number, ByteUnit] {
+  const magnitude = Math.floor(Math.log2(bytes === 0 ? 1 : bytes) / 10);
 
-  return [remainder, units[magnitude]];
+  return [Number.parseFloat((bytes / 2 ** (magnitude * 10)).toFixed(maxPrecision)), magnitude];
```

Comment:
Maybe we can use base 1024 for these functions? But this isn't wrong at all.

```suggestion
  const magnitude = Math.floor(Math.log(bytes === 0 ? 1 : bytes) / Math.log(1024));

  return [Number.parseFloat((bytes / 1024 ** magnitude).toFixed(maxPrecision)), magnitude];
```

Reply:
Oh yeah we could, good idea!